backend/python/tapa/floorplan.py

```diff
@@ -0,0 +1,248 @@
+import itertools
+import logging
+import os
+import sys
+from concurrent import futures
+from typing import Optional, Dict, Any, Callable, List, Union, Set, Tuple
+
+import tapa.autobridge
+from haoda.report.xilinx import rtl as report
+from tapa import util
+from tapa.task import Task
+from tapa.verilog import xilinx as rtl
+
+from .instance import Instance
+
+_logger = logging.getLogger().getChild(__name__)
+
+class InputError(Exception):
+  pass
+
+
+def get_cpp(cpp_dir: str, name: str) -> str:
```

Comment:
Instead of copy-and-pasting the code, let's pass `tapa.core.Program.get_cpp` as an argument (like `task_getter`), in case the directory hierarchy changes in the future. Longer term, if we ever get to refactor the code, we may need to define these small utilities in their own module, so other modules can import them without having to worry about cyclic dependency (my bad!).